src/uk/gov/defra/ffc/Tests.groovy

```diff
@@ -22,6 +22,23 @@ class Tests implements Serializable {
     }
   }
 
+  static def runZapScan(ctx, projectName, buildNumber, tag) {
+    def zapDockerComposeFile = 'docker-compose.zap.yaml'
+    if (ctx.fileExists("./$zapDockerComposeFile")) {
+      ctx.gitStatusWrapper(credentialsId: 'github-token', sha: Utils.getCommitSha(ctx), repo: Utils.getRepoName(ctx), gitHubContext: GitHubStatus.ZapScan.Context, description: GitHubStatus.ZapScan.Description) {
+        try {
+          // test-output exists if stage is run after 'runTests', take no risks and create it
+          ctx.sh('mkdir -p -m 666 test-output')
+          ctx.sh("docker-compose -p $projectName-$tag-$buildNumber -f docker-compose.yaml -f $zapDockerComposeFile run zap-baseline-scan")
```

Comment:
This is mainly my ignorance I think, but why are there two alternate compose files specified with the `-f` option here - in effect the default AND an alternative?

Reply:
Specifying the files like that allows the config for the service to be extended by each file. In this case, the env vars are set in `docker-compose.yaml` and do not need to be added in `docker-compose.zap.yaml`.